py-rattler/src/networking/mod.rs

```diff
@@ -30,7 +31,7 @@ pub fn py_fetch_repo_data<'a>(
     callback: Option<&'a PyAny>,
 ) -> PyResult<&'a PyAny> {
     let mut meta_futures = Vec::new();
-    let client = PyAuthenticatedClient::new();
+    let client = PyClientWithMiddleware::new(None);
```

Comment:
Is it desired that this change removes the auth middleware here?

Reply:
Yes I think so, `AuthenticatedClient` is now a Python class that just has a default middleware passed in Python

Reply:
i would say to go in line with `rattler.install` where by default we don't have any authentication and need to specify `client=AuthenticatedClient()` we should change this here

Reply:
added the ability to specify `client` in `py_fetch_repo_data`